src/Bicep.Core.UnitTests/Utils/TextByteArray.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Bicep.Core.UnitTests.Utils;
+
+/// <summary>
+/// A byte array for UTF-8 text
+/// </summary>
+public class TextByteArray
+{
+    private readonly ImmutableArray<byte> _bytes;
+
+    public TextByteArray(string text) => _bytes = Encoding.UTF8.GetBytes(text).ToImmutableArray();
+
+    public TextByteArray(byte[] bytes) => _bytes = bytes.ToImmutableArray();
+
+    public TextByteArray(ImmutableArray<byte> bytes) => _bytes = bytes;
+
+    public ImmutableArray<byte> Bytes => _bytes;
+
+    public byte[] ToArray() => _bytes.ToArray();
+
+    // Left as property to make it easier to use in the debugger
+    public string Text => Encoding.UTF8.GetString(_bytes.ToArray());
```

Comment:
We can replace `ToArray()` with `AsSpan()` to avoid allocation.

Reply:
Also if it's purely for debuggability, we could use debugger display attributes or similar mechanisms.

Reply:
@majastrz Debuggability is the main frustration, yes.  I did investigate debugger display et al in VS but it didn't pan out very well (and want to work in VS and VS Code).

Either way, this seems like a more direct and simple approach rather than storing a stream to a binary buffer that's really UTF-8 text.  But note that the current changes are for unit tests only (although I'd be for using in production code if it makes sense), and it's not storing the data as a string, it's still storing as the exact binary data we were given (which can be important at least for the tests).

Reply:
@shenglol Changed to AsSpan, thanks.